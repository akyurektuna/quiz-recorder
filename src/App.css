.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

button {
  align-items: center;
  width: 157px;
  height: 40px;
  background: #57647e;
  border: initial;
  border-radius: 4px;
  font-family: Inter,sans-serif;
  font-size: 16px;
  font-weight: 500;
  text-transform: capitalize;
  padding: 10px 11px 10px 18px;
  color: #fff;
  text-shadow: none;
  box-shadow: none;
}

.button1{
  padding-left: 4px;
  display:inline;
}
.button2{
  padding-left: 4px;
  display:inline;
}
.button3{
  padding-left: 4px;
  display:inline;
}
.fa.fa-car{
  color:red;
}

article .header{
  font-size: 16px;
}