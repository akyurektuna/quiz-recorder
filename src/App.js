import React from 'react';
import useMediaRecorder from './wmik/use-media-recorder';
import '../src/App.css';

function Player({ srcBlob, audio }) {
  if (!srcBlob) {
    return null;
  }

  if (audio) {
    return <audio src={URL.createObjectURL(srcBlob)} controls />;
  }

  console.log(URL.createObjectURL(srcBlob));
  if(srcBlob){
    window.parent.postMessage(URL.createObjectURL(srcBlob), '*');
  }
  

  return (
    <video
      src={URL.createObjectURL(srcBlob)}
      width={520}
      height={480}
      controls
    />
  );
}

function ScreenRecorderApp() {
  let {
    error,
    status,
    mediaBlob,
    stopRecording,
    getMediaStream,
    startRecording
  } = useMediaRecorder({
    recordScreen: true,
    blobOptions: { type: 'video/webm' },
    mediaStreamConstraints: { audio: true, video: true }
    
  });

  // function onRecordingFinished(){
  //     if(mediaBlob){
  //       window.parent.postMessage(mediaBlob, '*');
  //       console.log(URL.createObjectURL(mediaBlob));
  //     }
  // };

  

  return (
    <article>
      <div className="header">Screen recorder</div>
      {error ? `${status} ${error.message}` : status}
      <section>
        <div className="button1">
          <button
            type="button"
            onClick={getMediaStream}
            disabled={status === 'ready'}
          >
            Share screen
        </button>
        </div>
        <p className="button2">
          <button
            type="button"
            onClick={startRecording}
            disabled={status === 'recording'}
          >
            Start recording
        </button>
        </p>
        <p className="button3">
          <button
            type="button"
            onClick={stopRecording}
            //onClick={onRecordingFinished}
            disabled={status !== 'recording'}
            
          >
            <i class="far fa-stop-circle"></i>
          Stop recording
        </button>
        </p>


      </section>

      <Player srcBlob={mediaBlob} />
    </article>
  
  );
  
}



export default ScreenRecorderApp;