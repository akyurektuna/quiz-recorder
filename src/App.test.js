import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import App from './App'; //test etmek istedigini import et
//import { isTSAnyKeyword } from @babel/types;
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";


// Enzyme.configure({ adapter: new Adapter() })

it("renders without crushing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App> </App>, div);
})

it("renders without crashing", () =>{
  shallow(<App />);
});

it("renders header ", () => {
  const wrapper = shallow(<App />);
  const welcome = <h1>recorder app</h1>;
  expect(wrapper.contains(welcome)).toEqual(true);
});

it.skip('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

it("renders correctly with no error message", () => {
  const wrapper = mount();
  expect(wrapper.state("error")).toEqual(null);
});

it("renders correctly", () => {
  const tree = shallow(<App />);
  expect(toJson(tree)).toMatchSnapshot();
});