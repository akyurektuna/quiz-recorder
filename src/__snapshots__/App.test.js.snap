// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article>
  <div
    className="header"
  >
    Screen recorder
  </div>
  idle
  <section>
    <div
      className="button1"
    >
      <button
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Share screen
      </button>
    </div>
    <p
      className="button2"
    >
      <button
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Start recording
      </button>
    </p>
    <p
      className="button3"
    >
      <button
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <i
          class="far fa-stop-circle"
        />
        Stop recording
      </button>
    </p>
  </section>
  <Player
    srcBlob={null}
  />
</article>
`;
